### ISSUE TITLE
Globstar pattern returns an empty string as an element of the results array

### ISSUE THREAD

**Issue by mrmlnc** [isaacs/node-glob#519]
#### What the problem?

The `**` pattern returns an empty string as an element of the results array.

#### Steps to reproduce

```js
const ng = require('glob');

// MaxDepth option only for shortening the list of results
const entries = ng.globSync('**', { maxDepth: 1 });

console.dir(entries, { colors: true });
```

#### Actual behaviour

```js
[
  '',
  'tsconfig.json',
  'src',
  'package.json',
  'out',
  'node_modules',
  'index.js',
  'herebyfile.mjs',
  'fixtures',
  'README.md',
  'LICENSE'
]
```

#### Expected behavioud

```diff
[
- '',
  'tsconfig.json',
  'src',
  'package.json',
  'out',
  'node_modules',
  'index.js',
  'herebyfile.mjs',
  'fixtures',
  'README.md',
  'LICENSE'
]
```

**Comment by mrmlnc** [comment]
I think this is a very inconvenient solution for the end user. An empty string cannot be used in many standard methods for working with the file system.

```js
> fs.readdir('')
Uncaught:
TypeError [ERR_INVALID_CALLBACK]: Callback must be a function. Received undefined
    at __node_internal_captureLargerStackTrace (node:internal/errors:465:5)
    at new NodeError (node:internal/errors:372:5)
    at __node_internal_ (node:internal/validators:224:11)
    at makeCallback (node:fs:186:3)
    at Object.readdir (node:fs:1369:14) {
  code: 'ERR_INVALID_CALLBACK'
}

> fs.lstatSync('')
Uncaught Error: ENOENT: no such file or directory, lstat
    at Object.lstatSync (node:fs:1529:3) {
  errno: -2,
  syscall: 'lstat',
  code: 'ENOENT'
}
```

And every time the user will have to exclude this element from the results array.

And, yeap, makes sense with the options that you mentioned above.

**Comment by isaacs** [comment]
Hm.  Maybe I could have `dotRelative` default to true when the result is an empty string?

**Comment by isaacs** [comment]
@mrmlnc Can you check out https://github.com/isaacs/node-glob/pull/520 and see how that works for you?

**Comment by isaacs** [comment]
Someone else pinged me about the same thing, so just further validation that this is legitimately confusing.

Landed in 10.1.0, `''` will never be returned now, only `'.'`.

**Comment by mrmlnc** [comment]
Sorry to keep you waiting. I continue to think that this is extra information for the user in most cases, but I have a similar request in fast-glob (https://github.com/mrmlnc/fast-glob/issues/47) and I am also going to implement it in the near future.
